anvil-python/anvil/provider/local/commands.py

```diff
@@ -462,6 +464,23 @@ def join(
             )
         )
 
+    machines_res = subprocess.run(
```

Comment:
What about hiding this logic inside the `is_skip` method as MaasScaleJujuStep of Sunbeam is doing? So that the Step can be skipped or the machines_to_join are added to extra_args: `self.extra_args.extend("--to", ",".join(machines_to_join)`